### PR TITLE
Living on the edge

### PR DESCRIPTION
**Pull request by pranavkm**
Update SDK to the one with source generators

### CODE REVIEW

File: src/Identity/UI/src/Microsoft.AspNetCore.Identity.UI.csproj [hunk 1]
```diff
@@ -47,113 +43,22 @@
...
+  <Target Name="SetupRazorInputs" BeforeTargets="ResolveRazorGenerateInputs">
     <ItemGroup>
-      <_RazorGenerate Include="Areas\Identity\Pages\$(UIFrameworkVersion)\**\*.cshtml" />
+      <_RazorGenerate Include="Areas\Identity\Pages\V4\**\*.cshtml" />
 
       <RazorGenerate Include="@(_RazorGenerate)" Link="Areas\Identity\Pages\%(RecursiveDir)%(Filename)%(Extension)" />
     </ItemGroup>
   </Target>
 
-  <Target Name="BuildForUI" DependsOnTargets="SetupRazorInputs;RazorCompile" />
```
Comment (pranavkm) [inline src/Identity/UI/src/Microsoft.AspNetCore.Identity.UI.csproj:113]: FYI @javiercn. The `RazorCompile` target no longer runs for 6.0 (compilation of Razor files is part of Csc). I didn't spend a lot of time trying to continue supporting multitargeting here.

File: src/Tools/Microsoft.dotnet-openapi/test/dotnet-microsoft.openapi.Tests.csproj [hunk 2]
```diff
@@ -18,6 +18,7 @@
 
   <ItemGroup>
     <Reference Include="Microsoft.Build" ExcludeAssets="runtime" />
+    <Reference Include="NuGet.Frameworks" />
```
Comment (pranavkm) [inline src/Tools/Microsoft.dotnet-openapi/test/dotnet-microsoft.openapi.Tests.csproj:21]: @dougbu turns out the newer SDK expects support for the .NET 6.0 TFM. These tests evaluate MSBuild in-proc and we have to somehow squirrel an updated version to the test.

Comment (dougbu) [reply src/Tools/Microsoft.dotnet-openapi/test/dotnet-microsoft.openapi.Tests.csproj:21]: Not sure how to parse this @pranavkm. Without a PR description, I have no way of knowing how the pieces fit together, beyond this change and the one in eng/Dependencies.props being linked. So, what's happening here❔

File: src/Identity/UI/src/Microsoft.AspNetCore.Identity.UI.csproj (hunk 1 above)
Comment (javiercn) [reply src/Identity/UI/src/Microsoft.AspNetCore.Identity.UI.csproj:113]: I'm somewhat ok with this, but we are going to have to figure an alternative for the future. We are likely upgrading to Bootstrap5 and I'm not sure if we'll have to keep 4 around to ease migration.

**Review by javiercn — APPROVED** [review]
LGTM!

Please file an issue so that we discuss the Identity UI SxS support in the future.

File: src/Tools/Microsoft.dotnet-openapi/test/dotnet-microsoft.openapi.Tests.csproj (hunk 2 above)
Comment (pranavkm) [reply src/Tools/Microsoft.dotnet-openapi/test/dotnet-microsoft.openapi.Tests.csproj:21]: Sorry, let me rephrase. 

* The openAPI tests host an MSBuild workspace in process. https://github.com/dotnet/aspnetcore/blob/main/src/Tools/Microsoft.dotnet-openapi/src/Commands/BaseCommand.cs#L129-L137 which results in an MSBuild evaluation of the project using the SDK we have installed.
* The most recent SDK expects a version of NuGet.Framework that supports the 6.0 TFM.
* Prior to this change, the NuGet.Versioning package was brought in transitively by Microsoft.NET.Sdk.Test SDK that's added to all test projects:
```
"Microsoft.TestPlatform.ObjectModel/16.6.1": {
        "type": "package",
        "dependencies": {
          "NuGet.Frameworks": "5.0.0"
        },
```
* Using the 5.0 version of the package results in the MSBuild evaluation and the test to fail. 
* Adding the reference and updating the package reference to a newer version fixes the problem.

Comment (pranavkm) [reply src/Tools/Microsoft.dotnet-openapi/test/dotnet-microsoft.openapi.Tests.csproj:21]: Btw, here's the error prior to this change:

```
Threw error: The expression "[MSBuild]::GetTargetFrameworkIdentifier(net6.0)" cannot be evaluated. MSB0001: Internal MSBuild Error: A required NuGet assembly was not found. Expected Path: /datadisks/disk1/work/9FD308A7/w/A22A0907/e/.dotnet24379/sdk/6.0.100-preview.2.21118.6  /datadisks/disk1/work/9FD308A7/w/A22A0907/e/.dotnet24379/sdk/6.0.100-preview.2.21118.6/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.TargetFrameworkInference.targets
```

Comment (dougbu) [reply src/Tools/Microsoft.dotnet-openapi/test/dotnet-microsoft.openapi.Tests.csproj:21]: I see. So, we're basically working around out-of-date dependencies in the latest Microsoft.TestPlatform.ObjectModel packages. Ick but :ok:
